src/tools/snapshot.ts

```diff
@@ -183,8 +186,9 @@ const selectOption: Tool = {
   },
 };
 
-const screenshotSchema = z.object({
+const screenshotSchema = elementSchema.extend({
```

Comment:
This is confusing as the element ref becomes required. It should be optional, and it should control whether this is an element screenshot or a viewport screenshot.